test/OriginationControllerSTIRFRY.ts

```diff
@@ -0,0 +1,1062 @@
+    await whitelistCurrency2.wait();
+    // verify the currencies are whitelisted
+    const isWhitelisted = await originationHelpers.isAllowedCurrency(USDC.address);
+    expect(isWhitelisted).to.be.true;
+    const isWhitelisted2 = await originationHelpers.isAllowedCurrency(sUSDe.address);
+    expect(isWhitelisted2).to.be.true;
+
+    // admin whitelists pair for stirfry loans
+    await originationControllerSTIRFRY.setPair(USDC.address, sUSDe.address, 1e12, true);
+    // verify the pair is whitelisted
+    const key = ethers.utils.solidityKeccak256(["address", "address", "uint256"], [USDC.address, sUSDe.address, 1e12]);
+    const isPairWhitelisted = await originationControllerSTIRFRY.stirfryPairs(key);
+    expect(isPairWhitelisted).to.be.true;
+
+    // admin whitelists MockERC721 and vaultFactory on OriginationController
+    await originationHelpers.setAllowedCollateralAddresses([vaultFactory.address], [true]);
+
+    // verify the collateral is whitelisted
+    const isVaultFactoryWhitelisted = await originationHelpers.isAllowedCollateral(vaultFactory.address);
+    expect(isVaultFactoryWhitelisted).to.be.true;
+
+    const repaymentController = <RepaymentController>await deploy("RepaymentController", deployer, [loanCore.address, feeController.address]);
+
+    await repaymentController.deployed();
+    const updateRepaymentControllerPermissions = await loanCore.grantRole(REPAYER_ROLE, repaymentController.address);
+    await updateRepaymentControllerPermissions.wait();
+
+    const updateOriginationControllerPermissions = await loanCore.grantRole(
+        ORIGINATOR_ROLE,
+        originationControllerSTIRFRY.address,
+    );
+    await updateOriginationControllerPermissions.wait();
+
+    return {
+        originationHelpers,
+        originationControllerSTIRFRY,
+        repaymentController,
+        feeController,
+        USDC,
+        sUSDe,
+        vaultFactory,
+        vault,
+        lenderPromissoryNote: lenderNote,
+        borrowerPromissoryNote: borrowerNote,
+        loanCore,
+        user: deployer,
+        other: signers[1],
+        signers: signers.slice(2),
+        blockchainTime,
+    };
+};
+
+const createLoanTerms = (
+    payableCurrency: string,
+    collateralAddress: string,
+    {
+        durationSecs = BigNumber.from(360000),
+        principal = ethers.utils.parseEther("100"),
+        interestRate = BigNumber.from(1),
+        collateralId = "1",
+        deadline = 1754884800,
+        affiliateCode = ethers.constants.HashZero
+    }: Partial<LoanTerms> = {},
+): LoanTerms => {
+    return {
+        durationSecs,
+        principal,
+        interestRate,
+        collateralId,
+        collateralAddress,
+        payableCurrency,
+        deadline,
+        affiliateCode
+    };
+};
+
+const defaultSigProperties: SignatureProperties = {
+    nonce: 1,
+    maxUses: 1,
+};
+
+describe("OriginationControllerSTIRFRY", () => {
```

Comment:
It's worth adding a test where the `payableToVaultedCurrencyRatio` is a fraction or more complex than 1:1.

Reply:
Since `payableToVaultedCurrencyRatio` is a uint, it cant be a fraction. The example in the tests uses a 1e12:1 ratio for sUSDe:USDC.

`payableToVaultedCurrencyRatio` is trusted input from the contract owner. There are notes about this in the natspec for `setPair()` detailing how to calculate this ratio and what it is used for.